### PR TITLE
Validate pipeline bundles being tracked

### PR DESCRIPTION
**Pull request by lcarva**

### CODE REVIEW

File: .github/workflows/pre-merge-ci.yaml [hunk 1]
```diff
@@ -25,6 +25,14 @@ jobs:
     - name: Run checks
       run: make ci
 
+    - name: Validate added bundles
+      run: |
+        set -o errexit
+        set -o pipefail
+        set -o nounset
```
Comment (zregvart) [inline .github/workflows/pre-merge-ci.yaml:32]: Not sure we need this, doesn't hurt, so nitpickery

File: hack/validate-acceptable-bundles.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+set -o errexit
+set -o pipefail
+set -o nounset
+
+bundles_file='data/acceptable_tekton_bundles.yml'
+
+function list_pipeline_bundles() {
+    < "${1}" yq \
+        '."pipeline-bundles" | to_entries | .[] | .key + "@" + .value[].digest' | \
+        sort -u
```
Comment (zregvart) [inline hack/validate-acceptable-bundles.sh:34]: (nitpick, for fun) you could use [`unique`](https://mikefarah.gitbook.io/yq/operators/unique)

Something like:

```suggestion
    < "${1}" yq \
        '[."pipeline-bundles" | to_entries | .[] | .key + "@" + .value[].digest] | unique | .[]'
```

File: hack/validate-acceptable-bundles.sh [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+origin_bundles="$(list_pipeline_bundles <(curl -s "https://raw.githubusercontent.com/hacbs-contract/ec-policies/main/${bundles_file}"))"
+pr_bundles="$(list_pipeline_bundles "${bundles_file}")"
+new_bundles="$(comm -13 <(echo "${origin_bundles}") <(echo "${pr_bundles}"))"
```
Comment (zregvart) [inline hack/validate-acceptable-bundles.sh:39]: TIL `comm`, neat

File: hack/validate-acceptable-bundles.sh [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+all_success=true
+for ref in ${new_bundles}; do
+    # Verify the image is accessible
+    skopeo inspect --raw "docker://${ref}" > /dev/null
```
Comment (zregvart) [inline hack/validate-acceptable-bundles.sh:44]: Do we want to error out if it isn't, or continue?

```suggestion
    skopeo inspect --raw "docker://${ref}" > /dev/null || { all_success=false; continue; }
```

**Review by zregvart — APPROVED** [review]
:+1: for the emoji!

File: hack/validate-acceptable-bundles.sh (hunk 2 above)
Comment (lcarva) [reply hack/validate-acceptable-bundles.sh:34]: Yes! Also need `sort`

File: hack/validate-acceptable-bundles.sh (hunk 4 above)
Comment (lcarva) [reply hack/validate-acceptable-bundles.sh]: Good call. Updated with a similar approach

File: .github/workflows/pre-merge-ci.yaml (hunk 1 above)
Comment (lcarva) [reply .github/workflows/pre-merge-ci.yaml]: Removed it. Thanks

**Pull request merged**